modules/connectivity/locals.tf

```diff
@@ -178,6 +178,13 @@ locals {
     local.deploy_virtual_network_gateway[location] &&
     hub_network.config.virtual_network_gateway.config.gateway_sku_vpn != local.empty_string
   }
+  deploy_azure_firewall_policy = {
+    for location, hub_network in local.hub_networks_by_location :
+    location =>
+    local.deploy_hub_network[location] &&
+    hub_network.config.azure_firewall.enabled &&
+    length(try(local.custom_settings.azurerm_firewall["connectivity"][location].firewall_policy_id, "")) == 0
```

Comment:
Does this length comparison work when the try fails? E.g. if the `firewall_policy_id` is not set

Reply:
In short, yes as the `try()` is there to determine when `firewall_policy_id` isn't set, and then returns `""` which evaluates to length = `0`